src/plugins/cft/api.js

```diff
@@ -0,0 +1,244 @@
+                data: {
+                    bankInfo: true,
+                    phone:    true
+                }
+            }
+        },
+    });
+
+    assertResponseSuccess(response);
+
+    return response.body.data;
+}
+
+/**
+ * @returns {Promise.<Array>}
+ */
+async function fetchWallets() {
+    const response = await fetchJson(makeApiUrl("wallets"), {
+        log:                 logFetches,
+        headers:             defaultHeaders,
+        sanitizeRequestLog:  {},
+        sanitizeResponseLog: {},
+    });
+
+    assertResponseSuccess(response);
+
+    return response.body.hasOwnProperty('data') && response.body.data.hasOwnProperty('wallets')
+        ? response.body.data.wallets
+        : [];
+}
+
+/**
+ * @param dateFrom
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactions(dateFrom) {
+    let isFirstPage  = true;
+    let transactions = [];
+
+    const lastSyncTime = new Date(dateFrom).getTime();
+    const pagination   = {
+        limit:  20,
+        offset: 0,
+        total:  null,
+    };
+
+    console.log('last sync: ' + lastSyncTime + ' (' + dateFrom + ')');
+
+    while (isFirstPage || pagination.offset < pagination.total) {
+        const data = await fetchTransactionsInternal(pagination.limit, pagination.offset);
+
+        if (data.part.offset !== pagination.offset) { // банк ограничивает выборку
+            break;
+        }
+
+        if (isFirstPage) {
+            pagination.total = data.part.totalCount;
+            isFirstPage      = false;
+        }
+
+        const loadNext = data.data
+                             .filter((transaction) => transaction.itemType === 'OPERATION' && ('money' in transaction))
+                             .every((transaction) => {
+                                 const isActual = transaction.date > lastSyncTime;
+
+                                 isActual && transactions.push(transaction);
+
+                                 return isActual;
+                             });
+
+        if (loadNext) {
+            pagination.offset += pagination.limit;
+        } else {
+            break;
+        }
+    }
+
+    return transactions;
+}
+
+/**
+ * @param limit
+ * @param offset
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactionsInternal(limit, offset) {
+    const urlParams = {
+        'limit':  limit,
+        'offset': offset,
+    };
+
+    const response = await fetchJson(makeApiUrl("hst", urlParams), {
+        log:                 logFetches,
+        headers:             defaultHeaders,
+        sanitizeRequestLog:  {},
+        sanitizeResponseLog: {},
+    });
+
+    assertResponseSuccess(response, [
+        'OK',
+        'OK_SYNC',
+    ]);
+
+    return response.body;
+}
+
+/**
+ * @param response
+ * @param allowedStatuses
+ */
+const assertResponseSuccess = (response, allowedStatuses = ['OK']) => {
+    console.assert(
+        response.status === 200 && allowedStatuses.indexOf(response.body.status) > -1,
```

Comment:
IMO interpreting indexOf result is better to be strict:
in case of `does not contain` use `items.indexOf(item) === -1`
in case of `contains` use `items.indexOf(item) !== -1`
instead of `indexOf(item) >= 0`, `indexOf(item) > -1`, `indexOf(item) < 0` etc